### PR TITLE
FEATURE NMH-168: Hide licensed models from spark-nlp search [skip-test]

### PR DESCRIPTION
**Pull request by KshitizGIT**
Jira Ticket [NMH-168](https://johnsnowlabs.atlassian.net/browse/NMH-168)

Requires this PR to work https://github.com/JohnSnowLabs/models-hub/pull/158

[NMH-168]: https://johnsnowlabs.atlassian.net/browse/NMH-168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by maziyarpanahi — APPROVED** [review]

**Review by pabla — COMMENTED** [review]
Let's make it simpler.

We can add a condition to the `toSearchString` method:
```js
  if (ORIGIN) {
    searchParams.append('origin', ORIGIN);
  }
```
And we don't need to modify the App and Sidebar components.

File: docs/_frontend/webpack.config.js [hunk 1]
```diff
@@ -111,6 +111,7 @@ module.exports = (env, argv) => {
         SEARCH_ORIGIN:
           process.env.SEARCH_ORIGIN ||
           'https://search.modelshub.johnsnowlabs.com',
+        ORIGIN: process.env.ORIGIN || 'https://sparknlp.org',
```
Comment (pabla) [inline docs/_frontend/webpack.config.js:114]: I think we could remove a default value from here.

**Comment by KshitizGIT** [comment]
> Let's make it simpler.
> 
> We can add a condition to the `toSearchString` method:
> 
> ```js
>   if (ORIGIN) {
>     searchParams.append('origin', ORIGIN);
>   }
> ```
> 
> And we don't need to modify the App and Sidebar components.

If we just modify the `toSearchString` , the origin=https://sparknlp.org will appear in the browser url .

**Comment by pabla** [comment]
> @pabla , If we just modify the `toSearchString` , the origin=https://sparknlp.org will appear in the browser url . I think it would be better to hide origin=https://sparknlp.org from the url fragments in the browser

Yes, you are right, the origin shouldn't be visible in the browser address.
But I would like to keep all things related to origin in one place.
What if we introduce a new parameter to the `toSearchString` method and don't include `ORIGIN` when we generate an URL for the browser address?

```js
window.history.pushState(
  nextParams,
  window.title,
  '/models' + toSearchString(nextParams, false)
);
```

File: docs/_frontend/webpack.config.js (hunk 1 above)
Comment (pabla) [reply docs/_frontend/webpack.config.js:114]: I would prefer to keep the frontend code the same as we have in the `johnsnowlabs` repo.

Comment (KshitizGIT) [reply docs/_frontend/webpack.config.js]: Remove default value

**Review by pabla — APPROVED** [review]

**Pull request merged**